### ISSUE TITLE
Presentation processing event

### ISSUE THREAD

**Issue by Artemkaaas** [hyperledger/aries-framework-javascript#1379]
As a verifier application, I would show a loader when the proof is received and is in processing (because it takes several seconds to query ledgers). Is there a way to detect that right now?
 
If I find the proper place, AFJ moves the presentation record to the `presentation-received` state only after the proof is fully processed.  We can make one of the changes in order to achieve desirable behavior?
* Add a new extra state (for instance `PresentationProcessing` which is not part of correspondent Aries RFC) into the `PresentationExchange` record and set this state right at the beginning of the Presentation processing function. 
* Set the `PresentationReceived` state at the beginning of the function to process the presentation. The behavior will differ from other message-processing functions, so we will need to update them as well. Is it breaking change?

**Comment by TimoGlastra** [comment]
Yeah we should improve this behaviour. For the states we try to match the states of the RFCs, but we don't necessarily have to do that (I think we already have .e.g declined). 

I think I'd like adding a new state more,  However It would make sense to me that the state is set immediatley to presentation received and the verification is more a side effect. we could e.g. add an extra method `verifyPresentation` that will be automatically called when auto accept is true, but otherwise the processPresentation will just receive and store the presentation, and not verify it.
